airflow/providers/amazon/aws/operators/appflow.py

```diff
@@ -0,0 +1,545 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+import copy
+import json
+import sys
+from datetime import datetime, timedelta, timezone
+from time import sleep
+from typing import TYPE_CHECKING, List, Optional, cast
+
+if sys.version_info >= (3, 8):
+    from functools import cached_property
+else:
+    from cached_property import cached_property
+
+from airflow.exceptions import AirflowException
+from airflow.models import BaseOperator
+from airflow.operators.python import ShortCircuitOperator
+from airflow.providers.amazon.aws.hooks.appflow import AppflowHook
+from airflow.providers.amazon.aws.utils import datetime_to_epoch_ms, get_airflow_version
+
+if TYPE_CHECKING:
+    from mypy_boto3_appflow.type_defs import (
+        DescribeFlowExecutionRecordsResponseTypeDef,
+        DescribeFlowResponseTypeDef,
+        ExecutionRecordTypeDef,
+        TaskTypeDef,
+    )
+
+    from airflow.utils.context import Context
+
+EVENTUAL_CONSISTENCY_OFFSET: int = 15  # seconds
+EVENTUAL_CONSISTENCY_POLLING: int = 10  # seconds
+SUPPORTED_SOURCES = {"salesforce", "zendesk"}
+
+
+class AppflowBaseOperator(BaseOperator):
+    """
+    Amazon Appflow Base Operator class (not supposed to be used directly in DAGs).
```

Comment:
WDYT about setting `__all__` in this module so this can't _actually_ be imported externally if the idea is you really don't want users using it?

Although, I can easily be convinced to not inhibit using this if folks want to create their own operator subclassing this one too so I don't have a strong opinion on this.

Reply:
It is a good point indeed. But I would prefer to keep it accessible and extensible by relying on name conversion + docstring that clarifies that it is just a base operator that should be extended before use.

Is it ok @josh-fell?